com.unity.netcode.gameobjects/Editor/NetworkObjectEditor.cs

```diff
@@ -101,4 +101,28 @@ public override void OnInspectorGUI()
             }
         }
     }
+
+    /// <summary>
+    /// This assures that a user cannot place a GameObject with a NetworkObject component under a GameObject
+    /// with a NetworkManager component
+    /// </summary>
+    [InitializeOnLoad]
+    public static class NetworkObjectHierarchyMonitor
+    {
+        static NetworkObjectHierarchyMonitor()
+        {
+            EditorApplication.hierarchyChanged += OnHierarchyChanged;
+        }
+
+        private static void OnHierarchyChanged()
+        {
+            // Get all NetworkObjects in the Hierarchy
+            var allNetworkObjects = Resources.FindObjectsOfTypeAll<NetworkObject>();
```

Comment:
I'd imagine `EditorApplication.hierarchyChanged` to get called too often and `Resources.FindObjectsOfTypeAll<NetworkObject>()` looks a little expensive to me.

Is there any other method/callback to do this check on parent change? I guess `OnTransformParentChanged()`/`OnTransformChildrenChanged()` are not getting called in the Editor in EditMode, right?

Reply:
also `Object.FindObjectsOfType` vs `Resources.FindObjectsOfTypeAll` performance? 👀